### PR TITLE
feat(SearchResultListItems): Apply consistent styling

### PR DESCRIPTION
**Pull request by rsmarathe**
## Hey, I just made a Pull Request!

1. Add line clamping in `CatalogSearchResultListItem`
2. Fix alignment of `AdrSearchResultListItem`. 
3. Fix size and font in `AdrSearchResultListItem` to match other `SearchResultListItem`

Before 
image

After 
image

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by kuangp — APPROVED** [review]
ADR changes looks good!

File: plugins/adr/src/search/AdrSearchResultListItem.tsx
Comment (tudi2d) [inline plugins/adr/src/search/AdrSearchResultListItem.tsx:107]: Optional follow-up: I saw that there is `front-matter` used in the ADR plugin to parse MD. Looking at the search result & how the raw MD is displayed - would it be possible to render the MD in Search as well or would this lead to performance issues?

**Review by tudi2d — APPROVED** [review]
Awesome - thanks for addressing this @rsmarathe!

**Comment by tudi2d** [comment]
Adding @backstage/discoverability-maintainers as this affects search UX. Let's get a review from them & then merge it:)

**Comment by kuangp** [comment]
👉 @backstage/discoverability-maintainers can you take a final look when you get a chance?

**Review by djamaile — APPROVED** [review]

**Pull request merged**
